Compatibility with Arduino 1.6.0

patrickelectric opened the pull request:
Now the IDE can compile and use Arduino version 1.6.0
TODO: Need to test upload with real boards

patrickelectric committed:
First compilation with Arduino 1.6.0

With this commit, we can compile some boards using the Arduino 1.6.0.
Now we need to solve some problems with variants and the board menu :)

Signed-off-by: Patrick J.P <patrickelectric@gmail.com>

patrickelectric committed:
Board menu with processor choice

Now we have a submenu that shows the possible processors of the chosen
board.

Signed-off-by: Patrick J.P <patrickelectric@gmail.com>

patrickelectric committed:
Processor can be save in settings

Now the chosen processor can be stored in settings

patrickelectric committed:
Compiling almost all boards with 1.6.0

Compiling all boards except for Arduino pro mini.

Signed-off-by: Patrick J.P <patrickelectric@gmail.com>

patrickelectric committed:
Last corrections, everything right to merge.

Added frequency in board menu.
If exist, put the frequencies in the board menu that is valid

All boards are compiling in version 1.6.0 of Arduino

TODO: test all upload methods

Signed-off-by: Patrick J.P <patrickelectric@gmail.com>

patrickelectric committed:
Added frequency in board menu

If exist, put the frequencies in the board menu that is valid

Signed-off-by: Patrick J.P <patrickelectric@gmail.com>

patrickelectric committed:
All boards compiling

All boards are compiling in version 1.6.0 of Arduino
TODO: test all upload methods

Signed-off-by: Patrick J.P <patrickelectric@gmail.com>

patrickelectric committed:
removing KDevelop files

Signed-off-by: Patrick J.P <patrickelectric@gmail.com>

patrickelectric committed:
Reading boards.txt as UTF-8

Now we can read boards.txt that contain UTF-8 characters

Signed-off-by: Patrick J.P <patrickelectric@gmail.com>

mupuf commented:
Thanks Patrick!

I will have a look at your patches soon. However, could you fix the commits instead of adding more commits later to clean the cruft introduced by previous commits? I mean, the last 5 patches are unacceptable because they make it hard for people to study the changes.

If you wonder how to do this with git, you may be interested in looking into git rebase -i, that will allow you to squash, edit or reword commits :)

patrickelectric commented:
No problem, sorry for that !

mupuf commented:
If you keep on sending patches like that and if I like where you are going (which I so far do!), you may very well get commit rights :) So, bear with me while I annoy you with what seems like little issues :)

patrickelectric commented:
Ty for the comment @mupuf , I believe that everything is alright now =D

patrickelectric committed:
Added some comments and corrections

patrickelectric committed:
Added Doxygen to CMakeLists

Now we can use Doxygen

Signed-off-by: Patrick J.P <patrickelectric@gmail.com>

patrickelectric committed:
New README, grantlee folder update

Added some useful links and infomation about the project
Grantlee can be found in /usr/lib/grantlee or /usr/lib/grantlee/0.5 (QT4)

Signed-off-by: Patrick J.P <patrickelectric@gmail.com>

patrickelectric committed:
Added Portuguese translation

Signed-off-by: Patrick J.P <patrickelectric@gmail.com>

mupuf commented:
Thanks Patrick! I will have a look at them tonight! I did not notice you were so quick to fix your patches!

mupuf commented on IDEApplication.cpp:
```diff
@@ -138,8 +138,8 @@ void IDEApplication::initializePlugins()
-     }
+    {
+        qDebug() << "loading translator failed.";
+    }
```
This change is unrelated, this is something one should generally avoid.

No problem for this time though.

mupuf commented on env/Board.cpp:
```diff
@@ -60,31 +60,32 @@ void Board::listBoards()
             {
                 QString line = QString::fromLocal8Bit(boardsFile.readLine()).trimmed();
-	        //QString line = QString::fromLocal8Bit(boardsFile.readLine().simplified());
+                
```
There should not be any trailing spaces, you can configure your editor to do just that :)

If I merge this series, you will be encouraged to write another patch to remove them all :p

patrickelectric replied:
HOoo, sorry for that, I forgot to configure trailing spaces :/.

mupuf commented:
I merged it and fixed a few things related to the build system.

I also successfully tested the new code on my arduino Leonardo. In the process, I found out that we also need to require avrdude, so I updated Arch's package.

Looking forward to seeing more changes coming from you!

patrickelectric commented:
Ty @mupuf !
In this moment I am trying to use QT5 in place of QT4, maybe in this weekend its finished :pray:

mupuf commented:
Yepee! I started many times to do this but always stopped midway. Now that Qt5 is pretty well supported everywhere, I guess grantlee also supports qt5 so it sounds like a good time!

mupuf commented:
A bug introduced by this patch series:
- https://github.com/mupuf/arduide/issues/4

Oh, and I also realized that the documentation was not available anymore. If you have some time to fix these little issues, I would appreciate it :)

patrickelectric commented:
NP, I'm busy right now but I'll solve it.
I am compiling the QT5 version in my machine, in the place of grantlee it's possible to use qml of qt.
In the next week I will try to upload with some boards like uno, mini and nano.
